tools/grammar/src/main/java/org/opencypher/tools/grammar/TranslateGrammar.java

```diff
@@ -0,0 +1,152 @@
+ * limitations under the License.
+ *
+ * Attribution Notice under the terms of the Apache License 2.0
+ *
+ * This work was created by the collective efforts of the openCypher community.
+ * Without limiting the terms of Section 6, any Derivative Work that is not
+ * approved by the public consensus process of the openCypher Implementers Group
+ * should not be described as “Cypher” (and Cypher® is a registered trademark of
+ * Neo4j Inc.) or as "openCypher". Extensions by implementers or prototypes or
+ * proposals for change that have been documented or implemented should only be
+ * described as "implementation extensions to Cypher" or as "proposed changes to
+ * Cypher that are not yet approved by the openCypher community".
+ */
+package org.opencypher.tools.grammar;
+
+import static org.opencypher.grammar.Grammar.ParserOption.INCLUDE_LEGACY;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+
+import org.opencypher.grammar.Grammar;
+import org.opencypher.tools.g4processors.BNFProcessor;
+import org.opencypher.tools.g4processors.G4Processor;
+import org.xml.sax.SAXException;
+
+/** translate between supported grammar serialisations, via the Grammar object model */
+public class TranslateGrammar {
+
+    public static void main(String[] args) throws Exception
+    {
+        List<String> argList =  new ArrayList<>(Arrays.asList(args));
+        String inXml  = getArg("x", argList);
+        String inG4   = getArg("g", argList);
+        String inBnf  = getArg("b", argList);
+        String outXml = getArg("X", argList);
+        String outG4  = getArg("G", argList);
+        String outBnf = getArg("B", argList);
+        boolean includeLegacy = getBooleanArg("legacy", argList);
+        
+        if (argList.size() != 0) {
+            usage(args);
+            System.exit(1);
+        }
+        if (includeLegacy) {
+             System.setProperty( INCLUDE_LEGACY.name(), "true" );
+        }
+        // ought to check for duplicates - just take the first if they're silly
+        Grammar grammar = null;
+        Grammar.ParserOption[] parserOptions = Grammar.ParserOption.from( System.getProperties() );
+        int inputOptions = 0;
+        if (inXml != null) {
+            inputOptions++;
+            grammar = Grammar.parseXML( Paths.get( inXml ), parserOptions );
+        } else if ( inG4 != null) {
+            inputOptions++;
+            G4Processor g4Processor = new G4Processor();
+            grammar = g4Processor.processFile(inG4);
+        } else if (  inBnf != null) {
+            inputOptions++;
+            BNFProcessor bnfProcessor = new BNFProcessor();
+            grammar = bnfProcessor.processFile(inBnf);
+        }
+        if (inputOptions != 1) {

```

Comment:
This check doesn't actually work. The increments to `inputOptions` are performed in mutually exclusive code blocks (`else if`).